app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+    @user = @current_user
+  end
+
+  def create
+    @user = User.find(params[:user_id]) # Get the user from params
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params.merge(author: @user))
+
```

Comment:
- [ ] Great job so far in your implementation. However, When creating a comment the below error is thrown to the UI.
<img width="1440" alt="Screenshot 2023-11-02 at 21 25 10" src="https://github.com/noelfoka/Blog_app/assets/107192076/af84186b-2b5e-4dfb-8671-d33810e3532c">

  - This is because in your `create` action you specified the current user with the column `author` instead of `user`. To fix this issue so users are able to create comments without any errors kindly consider refactoring your code to look like below:
```
@comment = @post.comments.new(comment_params.merge(user: @user))
```